root-cert-stats/background.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+var rootCertStats = {};
+
+/*
+On an onHeadersReceived event, if there was a successful TLS connection
+established, fetch the root cert and look at its subject.
+
+If we haven't seen this subject before, add it. If we have, increment its stats.
+*/
+async function logRootCert(details) {
+  try {
+    let securityInfo = await browser.webRequest.getSecurityInfo(
+      details.requestId,
+      {"certificateChain": true}
+    );
+    if (securityInfo.state == "secure" || securityInfo.state == "weak") {
+      let rootName = securityInfo.certificates[securityInfo.certificates.length - 1].subject;
```

Comment:
I think that we may actually want to collect the `issuer` of the last certificate in the chain here (instead of the `subject`).

Reply:
I am under the impression (and testing seems to confirm this) that the certificate chain includes the trust root itself, which will (always?) be self-signed, so `issuer`===`subject`. In which case it's moot, but I thought `subject` seemed semantically better (subject is the entity associated with the certificate, so for a root cert, it's the trusted organization).

Reply:
The main reason why I though that we should collect the `issuer` here is that the table in the popup is titled "Certification Authorities", and (at least if I recall the right definition) CA is supposed to be the entity that "issues the certificates".

Reply:
> CA is supposed to be the entity that "issues the certificates"

Yes, it does issue certificates, but `subject` is the name of the entity identified by the certificate. So if the last cert in the chain is the trusted root (which I think it is) then the name of the authority owning it is found in the subject, not the issuer.

(In practice, as I say, it's generally moot, because usually root certs are self-issued, so subject===issuer. But if they are not, then we want to use the subject, not the issuer.)

I think. But if I'm wrong about any of this I'd be happy to be corrected.